chia/rpc/wallet_rpc_api.py

```diff
@@ -1300,19 +1322,21 @@ async def did_update_metadata(self, request) -> EndpointResult:
 
     async def did_get_did(self, request) -> EndpointResult:
         wallet_id = uint32(request["wallet_id"])
-        wallet: DIDWallet = self.service.wallet_state_manager.wallets[wallet_id]
+        wallet = self.service.wallet_state_manager.wallets[wallet_id]
+        assert isinstance(wallet, DIDWallet)
         my_did: str = encode_puzzle_hash(bytes32.fromhex(wallet.get_my_DID()), AddressType.DID.hrp(self.service.config))
         async with self.service.wallet_state_manager.lock:
-            coins = await wallet.select_coins(uint64(1))
-        if coins is None or coins == set():
-            return {"success": True, "wallet_id": wallet_id, "my_did": my_did}
-        else:
-            coin = coins.pop()
-            return {"success": True, "wallet_id": wallet_id, "my_did": my_did, "coin_id": coin.name()}
+            try:
```

Comment:
this is the new uniform interface for `select_coins()` for all wallets. It doesn't return an `Optional`, it throws on errors.